File: valid-mountain-array/valid-mountain-array.java
class Solution {
    public boolean validMountainArray(int[] arr) {
        int top = 0;
        for (int i = 1; i < arr.length; i++) {
            if (arr[i-1] >= arr[i]) {
                break;
            }

            top = i;
        }

        if (top == 0 || top == arr.length-1) {
            return false;
        }

        for (int i = top+1; i < arr.length; i++) {
            if (arr[i] >= arr[i-1]) {
                return false;
            }
        }

        return true;
    }
}

File: valid-mountain-array/valid-mountain-array.go
func validMountainArray(arr []int) bool {
    top := 0
	for i := 1; i < len(arr); i++ {
		if arr[i-1] >= arr[i] {
			break
		}

		top = i
	}

	if top == 0 || top == len(arr)-1 {
		return false
	}

	for i := top + 1; i < len(arr); i++ {
		if arr[i] >= arr[i-1] {
			return false
		}
	}

	return true
}